jetty-openid/src/main/config/etc/jetty-openid.xml

```diff
@@ -20,30 +20,29 @@
     </Arg>
     <Set name="executor"><Ref refid="ThreadPool"/></Set>
   </New>
-  <New id="OpenIdConfiguration" class="org.eclipse.jetty.security.openid.OpenIdConfiguration">
-    <Arg><Property name="jetty.openid.provider" deprecated="jetty.openid.openIdProvider"/></Arg>
-    <Arg><Property name="jetty.openid.provider.authorizationEndpoint"/></Arg>
-    <Arg><Property name="jetty.openid.provider.tokenEndpoint"/></Arg>
-    <Arg><Property name="jetty.openid.clientId"/></Arg>
-    <Arg><Property name="jetty.openid.clientSecret"/></Arg>
-    <Arg><Property name="jetty.openid.authMethod" default="client_secret_post"/></Arg>
-    <Arg><Ref refid="HttpClient"/></Arg>
-    <Call name="addScopes">
-      <Arg>
-        <Call class="org.eclipse.jetty.util.StringUtil" name="csvSplit">
-          <Arg><Property name="jetty.openid.scopes"/></Arg>
-        </Call>
-      </Arg>
-    </Call>
-  </New>
   <Call name="addBean">
     <Arg>
-      <New class="org.eclipse.jetty.security.openid.OpenIdLoginService">
-        <Arg><Ref refid="OpenIdConfiguration"/></Arg>
-        <Arg><Ref refid="BaseLoginService"/></Arg>
-        <Call name="setAuthenticateNewUsers">
-          <Arg type="boolean">
-            <Property name="jetty.openid.authenticateNewUsers" default="false"/>
+      <Ref refid="BaseLoginService"/>
+    </Arg>
+  </Call>
+  <Call name="addBean">
+    <Arg>
+      <New id="OpenIdConfiguration" class="org.eclipse.jetty.security.openid.OpenIdConfiguration">
+        <Arg><Property name="jetty.openid.provider" deprecated="jetty.openid.openIdProvider"/></Arg>
+        <Arg><Property name="jetty.openid.provider.authorizationEndpoint"/></Arg>
+        <Arg><Property name="jetty.openid.provider.tokenEndpoint"/></Arg>
+        <Arg><Property name="jetty.openid.clientId"/></Arg>
+        <Arg><Property name="jetty.openid.clientSecret"/></Arg>
+        <Arg><Property name="jetty.openid.authMethod" default="client_secret_post"/></Arg>
+        <Arg><Ref refid="HttpClient"/></Arg>
```

Comment:
Should use named Arg elements here to make the XML more readable and order independent

Reply:
Isn't this fragile because you could easily change one of the constructor argument names but forget to change this XML.

Reply:
You would get an error in that case as the name would not be found.